errors.go

```diff
@@ -0,0 +1,34 @@
+package flux
+
+import (
+	"strings"
+
+	"github.com/influxdata/flux/codes"
+)
+
+// Error is the error struct of flux.
+type Error struct {
+	Code codes.Code
+	Msg  string
```

Comment:
Should we add a comment that we expect Code and Msg to always be set and that Err is optional and only needed to wrap an existing error?

Also we should state that only the `Code` field should be inspected by calling code.

Reply:
Yea, I think so. Although message is also going to be optional. There may be situations where we can't add any additional context.